### PR TITLE
7.x branch initial support

### PR DESCRIPTION
**Pull request by icex**
Use the latest ruflin/elastica (master)
Fixes total hits 10k limit

I haven't tested it with a Elastic Search 6.x server but I assume it may cause some issues due to using elastica branch 7.x

### CODE REVIEW

**Comment by CauanCabral** [comment]
Unit tests are failing because of required stability of dependencies.

Should we relax that requirement or wait a stable release from `ruflin/elastica` package?

Have someway to support both ES versions (6.x and 7.x)?

File: src/Query.php [hunk 1]
```diff
@@ -558,11 +558,11 @@ protected function _execute()
     {
         $connection = $this->_repository->getConnection();
         $index = $this->_repository->getName();
-        $type = $connection->getIndex($index)->getType($this->_repository->getType());
+        $searchObject = $connection->getIndex($index)->createSearch($this->_repository->getType());
```
Comment (CauanCabral) [inline src/Query.php]: The current syntax of [`createSearch()`](https://elastica.io/api/latest/classes/Elastica.Index.html#method_createSearch) expect a `string|array|\Elastica\Query` as first parameter.
Maybe we can replace the instantiation of `$searchObject` with directly use [`Index::search`](https://elastica.io/api/latest/classes/Elastica.Index.html#method_search), just like:

```php
return $connection->getIndex($index)->search($query, $this->_searchOptions);
```

Comment (icex) [reply src/Query.php]: Hi, good point. I've changed it now. Hope it works

**Comment by othercorey** [comment]
@markstory should this be closed with your new changes?

**Comment by markstory** [comment]
Perhaps. These changes would enable the 3.x compatible version of the plugin to work with elasticsearch 7. The changes I've made are for the 4.x compatible version.

**Pull request closed without merging**